### PR TITLE
Feature/fasteffekt mandelbrot

### LINKED ISSUE #8: Mandelbrot

### PR DESCRIPTION
**Pull request by IR0NSIGHT**
fixes #8

### CODE REVIEW

**Comment by IR0NSIGHT** [comment]
fixes #8

File: src/mandelbrot.effekt [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+        var notDone = true;
+        var escape = 0;
+
+        while (notDone && z < 50) {
+          val zr = zrzr - zizi + cr;
+          zi = 2.0 * zr * zi + ci;
+
+          //preserve recalculation
+          zrzr = zr * zr;
+          zizi = zi*zi
```
Comment (phischu) [inline src/mandelbrot.effekt:34]:
```suggestion
          zizi = zi * zi
```

File: src/mandelbrot.effekt [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+
+          if (zrzr + zizi > 4.0) {
+            notDone = false;
+            escape = 1;
+          }
+          z = z + 1;
+        }
+
+        byteAcc = lshift( byteAcc, 1) + escape;
```
Comment (phischu) [inline src/mandelbrot.effekt:43]:
```suggestion
        byteAcc = lshift(byteAcc, 1) + escape;
```

File: src/mandelbrot.effekt [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+        bitNum = bitNum + 1;
+
+        if (bitNum == 8) {
+          sum = xor(sum,byteAcc);
```
Comment (phischu) [inline src/mandelbrot.effekt:47]:
```suggestion
          sum = xor(sum, byteAcc);
```

File: src/mandelbrot.effekt [hunk 4]
```diff
@@ -0,0 +1,97 @@
+
+import src/benchmark
+
+extern pure def xor(a: Int, b: Int): Int = "a ^ b"
+
+extern pure def lshift(a: Int, b: Int): Int = "a << b"
+
+def Mandelbrot(size: Int) = {
+  def brot(size: Int): Int = {
```
Comment (phischu) [inline src/mandelbrot.effekt:9]:
```suggestion
  def mandelbrot(size: Int): Int = {
```

File: src/mandelbrot.effekt [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+      }
+    }
+
+    return sum;
+  }
+
+
+
+  def benchmark() = {
+    return brot(size)
```
Comment (phischu) [inline src/mandelbrot.effekt:65]:
```suggestion
    return mandelbrot(size)
```

File: src/mandelbrot.effekt [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+        } else if (x == size - 1) {
+          byteAcc = lshift(byteAcc, (8 - bitNum));
+          sum = xor(sum, byteAcc);
+          byteAcc = 0;
+          bitNum = 0;
+        }
+      }
+    }
+
+    return sum;
```
Comment (phischu) [inline src/mandelbrot.effekt:59]:
```suggestion
    sum
```

File: src/mandelbrot.effekt [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+      (result == 50);
+    } else if ( size == 1) {
+      (result == 128);
+    } else {
+      println("no known result for this problemsize:"++show(size)++" result="++show(result));
+      false;
+    }
+  }
+
+  return innerBenchmarkLoop(1){benchmark}{verifyResult}
```
Comment (phischu) [inline src/mandelbrot.effekt:81]:
```suggestion
  innerBenchmarkLoop(1){benchmark}{verifyResult}
```

File: src/mandelbrot.effekt [hunk 8]
```diff
@@ -0,0 +1,97 @@
...
+  Mandelbrot(1);
+  Mandelbrot(750);
+  Mandelbrot(500);
+
+  //unit tests
+  if (xor(123,456) != 435) {
+    panic("XOR operation is faulty!")
+  }
+
+   if (lshift(123,456) != 31488) {
```
Comment (phischu) [inline src/mandelbrot.effekt:94]:
```suggestion
  if (lshift(123,456) != 31488) {
```

File: src/mandelbrot.effekt [hunk 9]
```diff
@@ -0,0 +1,97 @@
...
+    } else {
+      println("no known result for this problemsize:"++show(size)++" result="++show(result));
+      false;
+    }
+  }
+
+  return innerBenchmarkLoop(1){benchmark}{verifyResult}
+}
+
+def main() = {
```
Comment (phischu) [inline src/mandelbrot.effekt:84]: Hmm, do we want a main function in each benchmark?

**Review by phischu — APPROVED** [review]
Excellent!

File: src/mandelbrot.effekt (hunk 9 above)
Comment (IR0NSIGHT) [reply src/mandelbrot.effekt:84]: it makes it very easy to run each benchmark (also for a bash script later to verify correctness) so i will be keeping them for now.

**Pull request merged**